var/spack/repos/builtin/packages/go/package.py

```diff
@@ -104,14 +104,7 @@ def install(self, spec, prefix):
                 shutil.copy2(f, os.path.join(prefix, f))
 
     def setup_environment(self, spack_env, run_env):
-        try:
-            spack_env.set('GOROOT_FINAL', self.spec.prefix)
-            spack_env.set('GOROOT_BOOTSTRAP', self.spec['go-bootstrap'].prefix)
-        except KeyError:
-            # Needed because we try to get a build-only
-            # dependency, and this may fail during module 
-            # file creation
-            pass
+        spack_env.set('GOROOT_FINAL', self.spec.prefix)
```

Comment:
*Ah ha!*  or *Doh!*  [choose one].

I'll run a build with these in place and let you know how it goes.